### ISSUE TITLE
Balancing and Playtesting

### ISSUE THREAD

**Issue by Zet0rz** [Zet0rz/nzombies#42]
Balance changes might need to be taken before workshop release. Specifically, these curves are the ones we can balance:
- Zombie Speed
- Zombie Health
- Amount of zombies
- Zombie max and min speeds as well as variance in the speeds
- Player sprint and walk speeds
- Player sprint stamina
- Player health regen times and delays
- Perk damages and radiuses (Electric Cherry reload and PhD Flopper explosion)
- Chances for Teddy bear in box, and chances for power up drops
- Grenade throw force, damage, fuse timer
- Revive time (without Quick Revive), range

**Comment by Zet0rz** [comment]
I just recently changed the health curves to be +10% health each round, rather than a much lower and even declining curve. The original in CoD works with 100 base health, +100 each round until round 10, then +10% (steeper curve than what we have in nZombies)

**Comment by minin43** [comment]
Hey Zet0rz, I'm Deathking on steam. I'm close to being ready to do some play testing with the maps I've made, but the game keeps crashing whenever all players go down (both client and server). I don't know know why it's happening, but I don't know if it's caused by the gamemode code itself or something wrong with an entity.

**Comment by lolleko** [comment]
I think i encountered the same bug, but just once. We will have to debug the issue. It seems to be dedicated server only, which is pretty weird. Could be related to the new targeting system.

**Comment by minin43** [comment]
No, it was for sure happening solo, when I was testing the monkey bomb/1-inch punch. I had every entity on the map tho, so idk.

**Comment by minin43** [comment]
Also, on the topic of the sprinting:

Please increase sprint time by ~1 second. Maybe .5 seconds. 

Don't know if this is a bug or feature, but continuing to hold down sprint after you run out of stamina slows you down enough that it's slower than regular walking.

**Comment by func-octo** [comment]
Stamina on players should be increased 25 - 33% so we can run more than 10 feet.
Walk speed should be 5 - 10% more and sprint speed should be 1.8x walk speed.

**Comment by Zet0rz** [comment]
I was thinking of changing walk speed to 200 (currently 150). Sprint is 300. And I agree with a bit more stamina.

**Comment by minin43** [comment]
Go ahead and change it and I'll see what effect it has on the gameplay. But please fix the issue of "Don't know if this is a bug or feature, but continuing to hold down sprint after you run out of stamina slows you down enough that it's slower than regular walking."

**Comment by Zet0rz** [comment]
@minin43 The sprint module just sets your sprint speed to that of ply:GetWalkSpeed() which is literally your non-sprint speed. It might seem slower, but unless you have an addon that messes with it and/or your guns have a weight slowing system, it will be that of your walk speed and not slower.

**Comment by minin43** [comment]
Ah, CW2.0 has a weight slowing system. Could you just set it to what the default walkspeed is and not use that function? It'd be quite a bitch to go through and for EVERY gun, remove the weight system code. If not, I can work with it.

**Comment by Zet0rz** [comment]
I'll look into a way of incorporating this speed into the walk speed; it would be cool if you actually do slow down from heavy weapons. But shouldn't this not slow you down further, but rather actually make you faster? (Since I bypass the weapon mod's slow?)

**Comment by minin43** [comment]
In retrospect, you are correct. But this is coding, and nothing ever works as intended.

**Comment by Zet0rz** [comment]
Yeah I see the code doing that. It's because it sets the max possible speed to that of the run speed, but minus the gun's speed removal. Since the sprint speed is set down to walk speed (the only thing that can prevent sprinting), it will lower your speed below even that. But as soon as you let go, your sprint speed is set back up meaning your _max_ speed will not be lower than your walk.

I can fix this by overwriting the hook CW2 adds to instead use ply:GetMaxSprintSpeed() which we use to save the original value.

**Comment by Zet0rz** [comment]
Alright sprint changed:
**\- Walk Speed changed from 150 to 200**
**\- Stamina Loss changed from 2.5 to 2**
**\- CW2 Weight Speed now works with our sprint system (I overwrote it)**

**Comment by minin43** [comment]
Sweet sweet. Will definitely see how this affects gameplay.

**Comment by minin43** [comment]
Update on my thoughts towards balancing:

-Zombies are currently fine as they are.
-Player stats are fine as they currently are.
-For perks:
   -PhD flopper could use a larger explosion radius
   -Electric Cherry could use a bigger zapping radius
   -Double tap is too strong and should be reduced by at least 20% (a lot of guns also fire so quickly the sound bugs out, fooling players into thinking the gun isn't firing)
   -The vulture's aid perk's ammo drop lots of the time will not work with CW2.0 ammunition (BUG)
   -MP Quick Revive (and just reviving in general) could use an animation
   -Everything else is fine as it is
-Teddy bear chance needs to be decreased. It happens too often and it's quite obnoxious.
-The recent grenade radius increase needs to be reverted, as I often get close to killing myself with how big the range is (if you throw it straight up/down, you have to sprint away to get to safety, and even then it still might damage you)

**Comment by Zet0rz** [comment]
Balance changes based on @minin43 suggestions made in commit 9f99e5a675b9e26b299dc00f48b3c0d54978c379
Keep on giving the feedback :D

**Comment by minin43** [comment]
The gamemode feels too easy. Even in small, tightly quartered maps (not only do the zombies get stuck frequently, but), I can easily survive to round 16+ no problem, and at that point (when I have 2 PaP guns and all 4 perks) it's a matter of "how long do I want to keep on playing because I'm getting bored."

Can we increase zombie speed/life per round?

**Comment by Zet0rz** [comment]
We could increase their health curve a bit, maybe make it 100% like the original with +100 health every round until round 10, from where it's +10% (steeper than now, it's now +10% every round). We can also change start health from 75 to 100. I want to slow down zombie speed curves in general to match CoD better, but I can make it increase a bit faster. Thoughts?

**Comment by Zet0rz** [comment]
Try using the convars to increase the difficulty in terms of health. Set their base health to 100 and see what difference it does. Then try changing the 1.1 to another multiplier (per round multiplier). Tell us which seemed better.

**Comment by ScriptnLua** [comment]
Off-Topic: @Zet0rz how about this as a background? I think it could use some touching up, but what do you think? http://steamcommunity.com/sharedfiles/filedetails/?id=677608486

**Comment by func-octo** [comment]
Since BO3 gives the players 100 points for survivng the round, could you add something like that here? It helps in the long run, and keeps the player playing since they get rewarded after each round (Maybe you could make it 150 or 200 points for survivng special rounds?).

**Comment by Zet0rz** [comment]
> Since BO3 gives the players 100 points for survivng the round

The game doesn't give players 100 points for surviving a round. It gives 100 XP towards the levelling system that is outside the actual game/rounds. All numbers displayed in the middle of the screen are XP, points are only shown on the actual points HUD.

**Comment by func-octo** [comment]
Oh. I didn't know that xD There's xp in BO? I guess I never noticed...

**Comment by func-octo** [comment]
When did you start working on nzombies?

**Comment by Zet0rz** [comment]
About 2 weeks before this video's release: https://www.youtube.com/watch?v=hpkcwXonVp4

**Comment by func-octo** [comment]
Wow. That wasn't that long ago...

**Comment by func-octo** [comment]
Got another potential map with 4 sacrifices included. ttt_lost_temple_v2

**Comment by jaxjaxk** [comment]
192.99.21.138:27075 [AMT] NZombies | Pointshop2 | Jukebox | FastDL | 20+ maps
go here

**Comment by Punkinator7** [comment]
The dogs could be easier to deal with in solo, like limiting them to spawn 3 at a time, In multiplayer, it would be fine, but solo players will find it harder to survive later dog rounds, as it spawns them all at once.

**Comment by DjNathann** [comment]
So you only want 3 dogs to spawn instead of 6?

First of all Dogs have limited health. "They don't get more health like zombies"

Second of all The game would be to easy.

Third of all You could just modify the dog numbers.

**Comment by sh4d0w59** [comment]
Dogs spawn way too quickly and close atm.  Personally, I'd like to see them spawn further away and spawn slightly similar to the original zombies because atm they seem to spawn too close.  Also, grenades deal too little damage to zombies even in the first few waves.

**Comment by dragontuar** [comment]
hey I'm building my own map as I have had a design that did not work in other zombie games that had buildable maps and I was wondering if there was documentation on the integration into hammer and info on what entity settings for spawning

my steam username is the same as I have here in GitHub
